### DISCUSSION TITLE
Table summary row

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#2929] [Ideas]
I have prepared table summary row feature:

![image](https://user-images.githubusercontent.com/3818874/175832723-e5ac4908-e2da-4c68-a134-96b1a74f230c.png)

If this is something you are interested in, I can make PR with this.

**Comment by danharrin** [comment]
It really depends on what the implementation looks like, could you send example code a Filament user would add to use this? We already have `getTableContentFooter()` for this

**Reply by studiowizjo** [reply]
At first I thought to add method on table, eg.

```php
$table->summaryRow() //or summary() 
```

but after some considerations, I think it would be more elastic to add it on field level. Thanks to this user could more easily provide strategy for summarization, eg:

```php
TextColumn('amount')
  ->summary('sum'), //this could be default, but some other options would be also possible, eg: avg, min, max

TextColumn('person')
  ->summary('from_to'), //this could be default, and is showed on screen above. If the letters from - to match, then it shows also next letter, eg: AB - AD, up to three letters. If there are all the same words in column, it shows only first letter

TextColumn('person')
  ->summary('count_unique'), //shows how many unique values are in column: 4 unique, 6 repeated

BooleanColumn('is_active')
  ->summary(), //as in the example on screen above

TextColumn('created_at')
  ->dateTime()
  ->summary(), //as in the example on screen above. For now I`m not really happy that summary stretches the column width, so it remains to be determined how to show it in shorter form. It could be splitted in 2 lines or show it something like: d-d.m.Y, but of course there are some things which should be taken into consideration
```

**Reply by danharrin** [reply]
I think that syntax is a little bit too "magic" for my liking, we could use closures inside `summary()` that accept `Collection $records` and return a string to render as the summary?

**Reply by studiowizjo** [reply]
I would like it anyway to make possible of using closures, but I think it would be good to have possibility to choose from some predefined, preety common summarizion strategies. Maybe this syntax would be better:

```php
TextColumn('amount')
  ->summary(SummaryStrategy::sum()),

TextColumn('person')
  ->summary(SummaryStrategy::fromTo()),

TextColumn('person')
  ->summary(SummaryStrategy::countUnique()),

BooleanColumn('is_active')
  ->summary(), //default strategy for found column type
```

What do you think?

**Reply by danharrin** [reply]
Sure, that also works
